hazelcast-jca/src/test/java/com/hazelcast/jca/AbstractDeploymentHzClientTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+* Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.hazelcast.jca;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+
+import com.hazelcast.core.Hazelcast;
+
+/**
+* base class for jca tests that rely on client connections to hazelcast
+*
+* @author asimarslan
+* @author juanavelez
+*/
+public class AbstractDeploymentHzClientTest {
+    // Start a Hazelcast Instance that will be used by the RAR to connect to in client mode 
+    static {
+        Hazelcast.newHazelcastInstance();
```

Comment:
I am assuming the same way it is done when using a Server instance via the JCA ResourceAdapterImpl. Since the test is really an integration test run on a JEE Container (in this case glassfish), the instance (whether a server or client) will be stopped/shutdown via the javax.resource.spi.ResourceAdapter::stop method

```
 stop

void stop()

    This is called when a resource adapter instance is undeployed or during application server shutdown. This is a shutdown notification from the application server, and this method is called by an application server thread. The application server thread executes in an unspecified context.

    During this method call, a ResourceAdapter JavaBean is responsible for performing an orderly shutdown of the resource adapter instance. Any exception thrown by this method call does not alter the processing of the application server shutdown or resource adapter undeployment that caused this method call. The application server may log the exception information for error reporting purposes. 
```
(BTW, this is modified code as the current code has a condition to not shutdown if the instance is client. Was that the purpose of the comment to make me realize the shutdown should also be for the client?)
```
    /* (non-Javadoc)
     * @see javax.resource.spi.ResourceAdapter#stop()
     */
    public void stop() {
        HazelcastInstance instance = hazelcastInstance;
        if (instance != null) {
            instance.getLifecycleService().shutdown();
        }
    }

```